src/renderer/components/dock/dock-tab.store.ts

```diff
@@ -34,14 +39,19 @@ export class DockTabStore<T = any> {
     });
   }
 
-  protected serializeData() {
+  protected serializeData(): Record<TabId, T> {
```

Comment:
There is no grantee given the implementation that the values will be complete. We shouldn't mislead future maintainers.

```suggestion
  protected finalizeData(): Record<TabId, Partial<T>> {
```